examples/assets_machine_learning/dagster_with_mlflow/dagster_with_mlflow/assets.py

```diff
@@ -0,0 +1,157 @@
+        df = df[~df.title.isna()]
+
+    return df
+
+
+from sklearn.model_selection import train_test_split
+from dagster import multi_asset, AssetOut
+
+
+@multi_asset(outs={"training_data": AssetOut(), "test_data": AssetOut()}, compute_kind='pandas')
+def training_test_data(hackernews_stories):
+    hackernews_stories = hackernews_stories
+    X = hackernews_stories.title
+    y = hackernews_stories.descendants
+    # Split the dataset to reserve 20% of records as the test set
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
+    return (X_train, y_train), (X_test, y_test)
+
+
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+import numpy as np
+
+
+@multi_asset(compute_kind='numpy',
+    outs={"tfidf_vectorizer": AssetOut(), "transformed_training_data": AssetOut()}
+)
+def transformed_train_data(training_data):
+    X_train, y_train = training_data
+    # Initiate and fit the tokenizer on the training data and transform the training dataset
+    vectorizer = TfidfVectorizer()
+    transformed_X_train = vectorizer.fit_transform(X_train)
+    transformed_X_train = transformed_X_train.toarray()
+    y_train = y_train.fillna(0)
+    transformed_y_train = np.array(y_train)
+    return vectorizer, (transformed_X_train, transformed_y_train)
+
+
+@asset(compute_kind='numpy')
+def transformed_test_data(test_data, tfidf_vectorizer):
+    X_test, y_test = test_data
+    # Use the fitted tokenizer to transform the test dataset
+    transformed_X_test = tfidf_vectorizer.transform(X_test)
+    transformed_y_test = np.array(y_test)
+    y_test = y_test.fillna(0)
+    transformed_y_test = np.array(y_test)
+    return transformed_X_test, transformed_y_test
+
+port = '5000'
+
+
+@asset(compute_kind='xgboost')
+def xgboost_comments_model(transformed_training_data, transformed_test_data):
+    transformed_X_train, transformed_y_train = transformed_training_data
+    transformed_X_test, transformed_y_test = transformed_test_data
+    # Train XGBoost model, which is a highly efficient and flexible model
+
+    
+    mlflow.xgboost.autolog()
+    mlflow.set_experiment("my-experiment")
+
+    xgb_regressor = xgb.XGBRegressor(
+        objective="reg:squarederror", eval_metric=mean_absolute_error, n_estimators=20
+    )
+    xgb_regressor.fit(transformed_X_train, transformed_y_train, eval_set=[(transformed_X_test, transformed_y_test)])
+
+    experiment_id = mlflow.get_experiment_by_name("my-experiment").experiment_id
+    run_id = mlflow.last_active_run().info.run_id
+    my_run = mlflow.get_run(run_id).to_dictionary()
+    my_metadata = my_run['data']['metrics']
+
+    cmd = f'mlflow ui --port {port}'
+    os.system(cmd)
+
+    url = f"http://127.0.0.1:{port}"
+    link = f"{url}/#/experiments/{experiment_id}/runs/{run_id}"
+    my_metadata["mlflow link"] = MetadataValue.url(link)
+    return Output(xgb_regressor, metadata= my_metadata)
+
+
+@asset(compute_kind='mlflow')
+def model_registry(xgboost_comments_model):
```

Comment:
In the current pipeline, we save the model to a file at the end of the execution of `xgboost_comments_model`, and then we load it from that file here and store it in the MLflow model registry.

Might we want to just put it into the MLflow model registry directly at the end of `xgboost_comments_model`?